### ISSUE TITLE
nested pseudo element not taking correct precedence

### ISSUE THREAD

**Issue by zjrosen1** [rofrischmann/react-look#116]
### Example Case 1: hover pseudo class called in the default root styling

**Expected Behavior:**
- root hover state: blue
- secondary hover state: yellow

**Result:** _Working_
- root hover state: blue
- secondary hover state: yellow

```javascript
styles() {
  return {
    backgroundColor: 'red',
    ':hover': {
      backgroundColor: 'blue',
    },
    'secondary=true': {
      backgroundColor: 'green',
      ':hover': {
        backgroundColor: 'yellow',
      },
    },
  }
}
```

### Example Case 2: hover pseudo class called after the secondary hover styles

**Expected Behavior:**
- root hover state: blue
- secondary hover state: yellow

**Result:** _Failed_ nested hover class should take precedence
- root hover state: blue
- secondary hover state: blue

```javascript
styles() {
  return {
    backgroundColor: 'red',
    'secondary=true': {
      backgroundColor: 'green',
      ':hover': {
        backgroundColor: 'yellow',
      },
    },
    ':hover': {
      backgroundColor: 'blue',
    },
  }
}
```

**Comment by rofrischmann** [comment]
Hmm. I don't really get why in example 2 it should take the `yellow` backgroundColor over the `blue`? Is this a standard that is defined somewhere?

**Comment by zjrosen1** [comment]
Here is a [JSBin](http://jsbin.com/peniwapuje/1/edit?html,css,output) with *Example 2* as you can see the result of what I believe should be happening. I'm treating secondary=true as a nested selector to the root. A deeper selected style should override its parents styles.

**Comment by rofrischmann** [comment]
Ah I see. Damn css :D I have to think about it as this is not as easy as it might sound like.

**Comment by rofrischmann** [comment]
Could solve this by introducing precedence rules but this would add a lot of complexity which we might not want to have. 
**But**: There could be an optional plugin to solve this!
I will try to hack sth. soon,  stay tuned

**Comment by rofrischmann** [comment]
I considered this for a long time now, but I do not know any performant solution that does not add hundreds of lines of code... 
I will close this for now. If there really is a need for this, we might reopen, but as it is possible to work around this I do not see why it'd make sense to deeper investigate.

**Comment by rofrischmann** [comment]
This actually gets implemented with the next version 0.8
